.github/workflows/copy-react-demo-s3.yml

```diff
@@ -0,0 +1,63 @@
+name: Copy React Demo to S3 bucket
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'sample-apps/react/react-video-demo/**'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+jobs:
+  build-and-copy:
+    runs-on: ubuntu-latest
+
+    env:
+      VITE_STREAM_API_KEY: ${{ vars.EGRESS_STREAM_API_KEY }}
+      VITE_STREAM_TOKEN: ${{ secrets.EGRESS_USER_TOKEN }}
+      VITE_STREAM_KEY: ${{ vars.STREAM_API_KEY_SAMPLE_APPS }}
+      VITE_STREAM_SECRET: ${{ secrets.STREAM_SECRET_SAMPLE_APPS }}
+      VITE_VIDEO_USER_ID: ${{vars.VIDEO_DEMO_USER_ID}}
+      VITE_VIDEO_USER_NAME: ${{vars.VIDEO_DEMO_USER_NAME}}
+      VITE_VIDEO_USER_TOKEN: ${{secrets.VIDEO_DEMO_USER_TOKEN}}
+      VITE_MAPBOX_GL_TOKEN: ${{secrets.VIDEO_DEMO_MAPBOX_GL_TOKEN}}
+
+    steps:
+     - name: Checkout
+       uses: actions/checkout@v3
+       with:
+          fetch-depth: 0
+
+     - name: Setup Node
+       uses: actions/setup-node@v3
+       with:
+          node-version: 18.x
+          cache: 'yarn'
+
+     - name: Install Dependencies
+       run: yarn install --immutable
+
+     - name: Build packages
+       run: NODE_ENV=production yarn build:react:deps
+
+     - name: Test packages
+       run: yarn test:ci:all
+
+     - name: Build React Demo
+       run: yarn build:react:demo
+    
+     - name: Setup AWS credentials
+       uses: aws-actions/configure-aws-credentials@v2
+       with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{vars.AWS_REGION}}
+
+     - name: Copy artifacts to AWS S3 Bucket
+       run: |
+          aws s3 sync sample-apps/react/react-video-demo/build s3://${{vars.S3_BUCKET_NAME}}/react-video-demo --delete
```

Comment:
I think you should make two different directories in S3, as from what I see, PR builds and master builds would push code to the same dir in the bucket. This might be a bad idea as broken code can be accidentally pushed to the production site.

Reply:
Ah yes, thank you makes sense. I will change that. I might even remove PR builds